echidna/core/fill_spectrum.py

```diff
@@ -8,427 +8,296 @@
-
-    times = []
-    for time_step in range(0, spectrum._time_bins):
-        time = time_step * spectrum._time_width + spectrum._time_low
-        times.append(time)
-
-    if 'AV' in spectrumname:
-        print "AV WEIGHTS ARE CURRENTLY UNAVAILABLE"
-        weights = _av_weights(times, T)
-    else:
-        weights = _scint_weights(times, T)
-
-    for event in chain:
-        if event.scintFit == 0:
+        fill = True
+        fill_kwargs = {}
+        if bipo and ds.GetEVCount() != 1:
+            # Only bipos with 1 ev survive bipo cut
             continue
-        energy = event.energy
-        position = math.fabs(math.sqrt((event.posx)**2 +
-                                       (event.posy)**2 + (event.posz)**2))
-        spectrum._raw_events += 1
-        for time, weight in zip(times, weights):
+        for var, extractor in extractors.iteritems():
+            if extractor.get_valid_root(mc):
+                fill_kwargs[extractor._name] = extractor.get_value_root(mc)
+            else:
+                fill = False
+                break
+        if fill:
             try:
-                spectrum.fill(energy, position, time, 1.)
+                spectrum.fill(**fill_kwargs)
+                spectrum._raw_events += 1
             except ValueError:
                 pass
 
-    return spectrum
-
 
-def fill_mc_ntuple_spectrum(filename, T, spectrumname="", spectrum=None):
+def fill_from_root(filename, spectrum_name="", config=None, spectrum=None,
+                   bipo=False, **kwargs):
     """**Weights have been disabled.**
-    This function fills in the ndarray of energies, radii, times and
-    weights. It takes the true quenched energies and positions of the
-    events from ntuple. In order to keep the statistics, the time
-    dependence is performed via adding weights to every event depending
-    on the time period. Both, studied time and Half-life must be
-    written in the same units.
+    This function fills in the ndarray (dimensions specified in the config)
+    with weights. It takes the parameter specified in the config from the
+    events in the root file.
 
     Args:
-      filename (str): The ntuple to study
-      T (float): The Half-life of a studied background
-      spectrumname (str, optional): A name of future spectrum. Not
+      filename (str): A root file to study
+      spectrum_name (str, optional): A name of future spectrum. Not
         required when appending a spectrum.
+      config (:class:`echidna.core.spectra.SpectraConfig` or string, optional):
+        The config or directory to the config for the spectrum.
+        Not requried when appending a spectrum.
       spectrum (:class:`echidna.core.spectra.Spectra`, optional):
         Spectrum you wish to append. Not required when creating a
         new spectrum.
+      bipo (bool, optional): Applies the bipo cut if set to True.
+        Default is False.
+      \**kwargs (dict): Passed to and checked by the dsextractor.
```

Comment:
@jwaterfield, A couple more docs issues I've noticed, starting with this one. There are still a couple of places with `\**` in front of `kwargs`. the `\` causes the whole line not to be included in the docs. I think it's fine just to write `kwargs` as the `**` is really an operator.